server/src/main/java/org/opensearch/search/startree/filter/provider/BoolStarTreeFilterProvider.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.startree.filter.provider;
+
+import org.opensearch.index.mapper.CompositeDataCubeFieldType;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.query.TermsQueryBuilder;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.startree.filter.DimensionFilter;
+import org.opensearch.search.startree.filter.DimensionFilterMergerUtils;
+import org.opensearch.search.startree.filter.StarTreeFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Converts {@link BoolQueryBuilder} into {@link StarTreeFilter}
+ */
+public class BoolStarTreeFilterProvider implements StarTreeFilterProvider {
+    @Override
+    public StarTreeFilter getFilter(SearchContext context, QueryBuilder rawFilter, CompositeDataCubeFieldType compositeFieldType)
+        throws IOException {
+        return processBoolQuery((BoolQueryBuilder) rawFilter, context, compositeFieldType);
+    }
+
+    private StarTreeFilter processBoolQuery(
+        BoolQueryBuilder boolQuery,
+        SearchContext context,
+        CompositeDataCubeFieldType compositeFieldType
+    ) throws IOException {
+        if (boolQuery.hasClauses() == false) {
+            return null;
+        }
+        if (boolQuery.minimumShouldMatch() != null) {
+            return null; // We cannot support this yet and would need special handling while processing SHOULD clause
+        }
+        if (boolQuery.must().isEmpty() == false || boolQuery.filter().isEmpty() == false) {
+            return processMustClauses(getCombinedMustAndFilterClauses(boolQuery), context, compositeFieldType);
+        }
+        if (boolQuery.should().isEmpty() == false) {
+            return processShouldClauses(boolQuery.should(), context, compositeFieldType);
+        }
+        return null;
+    }
+
+    private StarTreeFilter processNonBoolSupportedQueries(
+        QueryBuilder query,
+        SearchContext context,
+        CompositeDataCubeFieldType compositeFieldType
+    ) throws IOException {
+        // Only allow other supported QueryBuilders
+        if (!(query instanceof TermQueryBuilder) && !(query instanceof TermsQueryBuilder) && !(query instanceof RangeQueryBuilder)) {
```

Comment:
Isn't this already handled by `SingletonFactory.getProvider(query)` ?

Reply:
Keeping additional check here, as when we support more query types we might need to handle changes in bool query handling

Reply:
Makes sense. 
maybe add a constant like `Set<Class<? extends QueryBuilder>> SUPPORTED_NON_BOOL_QUERIES`

Reply:
Made the change